backend/account/subscription_loader.py

```diff
@@ -75,3 +76,27 @@ def load_plugins() -> list[Any]:
         logger.info("No subscription plugins found.")
 
     return subscription_plugins
+
+def etl_prerun_check(org_id):
+    subscription_plugin_available = os.environ.get('SUBSCRIPTION_PLUGIN_AVAILABLE', 'False') == 'True'
+    if not subscription_plugin_available:
+        return True
+
+    try:
+        from pluggable_apps.subscription.subscription_helper import SubscriptionHelper
+    except ModuleNotFoundError:
+        logger.error("Subscription plugin not found.")
+        return False
+
+    org_plans = SubscriptionHelper.get_subscription(org_id)
+    if not org_plans or not org_plans.is_active:
+        return False
+
+    if org_plans.is_paid:
+        return True
```

Comment:
Do we actually need this check here? anyway we're returning True at the end.

Reply:
This check is not a necessary step as you mentioned, but by adding a check case for paid case here we can have a early return and avoid checking for the below conditions. This improves performance in such cases by avoiding additional checks before final return.

Hope this clarifies @jagadeeswaran-zipstack